src/components/vr_cooperation/src/command/on_default_service_chosen_notification.cc

```diff
@@ -58,9 +58,16 @@ void OnDefaultServiceChosenNotification::Execute() {
 
   vr_hmi_api::OnDefaultServiceChosenNotification params;
   vr_hmi_api::ServiceMessage msg = message();
-  if (msg.has_params() && params.ParseFromString(msg.params())
-      && params.has_appid()) {
-    parent()->SetDefaultService(params.appid());
+  if (msg.has_params()) {
+    if(params.ParseFromString(msg.params())) {
+      if (params.has_appid()) {
+        parent()->SetDefaultService(params.appid());
+      } else {
+        parent()->ResetDefaultService();
+      }
+    } else {
+      LOG4CXX_WARN(logger_, "Could not get result from message");
+    }
   } else {
     parent()->ResetDefaultService();
```

Comment:
In case of no appID defined, default service has been reset and HU-native service is uses by default (c) RSDL_HMI.proto